google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/project/AppEngineProjectServiceImpl.java

```diff
@@ -0,0 +1,199 @@
+/*
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.intellij.appengine.project;
+
+import com.google.cloud.tools.intellij.appengine.cloud.AppEngineEnvironment;
+
+import com.intellij.facet.Facet;
+import com.intellij.facet.FacetManager;
+import com.intellij.openapi.module.Module;
+import com.intellij.openapi.project.Project;
+import com.intellij.packaging.artifacts.Artifact;
+import com.intellij.packaging.impl.artifacts.ArtifactUtil;
+import com.intellij.psi.xml.XmlFile;
+import com.intellij.psi.xml.XmlTag;
+import com.intellij.remoteServer.configuration.deployment.ArtifactDeploymentSource;
+import com.intellij.remoteServer.configuration.deployment.DeploymentSource;
+
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import org.jetbrains.idea.maven.project.MavenProject;
+import org.jetbrains.idea.maven.project.MavenProjectsManager;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Set;
+
+public class AppEngineProjectServiceImpl extends AppEngineProjectService {
```

Comment:
Sure. Default is fine in this case.

On Thu, Jun 30, 2016 at 10:42 AM, Etan Shaul <notifications@github.com>
wrote:

> In
> google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/project/AppEngineProjectServiceImpl.java
> <https://github.com/GoogleCloudPlatform/gcloud-intellij/pull/764#discussion_r69145079>
> :
>
> > +import com.intellij.packaging.impl.artifacts.ArtifactUtil;
> > +import com.intellij.psi.xml.XmlFile;
> > +import com.intellij.psi.xml.XmlTag;
> > +import com.intellij.remoteServer.configuration.deployment.ArtifactDeploymentSource;
> > +import com.intellij.remoteServer.configuration.deployment.DeploymentSource;
> > +
> > +import org.jetbrains.annotations.NotNull;
> > +import org.jetbrains.annotations.Nullable;
> > +import org.jetbrains.idea.maven.project.MavenProject;
> > +import org.jetbrains.idea.maven.project.MavenProjectsManager;
> > +
> > +import java.util.Collection;
> > +import java.util.Collections;
> > +import java.util.Set;
> > +
> > +public class AppEngineProjectServiceImpl extends AppEngineProjectService {
>
> hmm I can't really think of any other possible implementations at the
> moment. Do you prefer the *Default*... naming pattern instead? I actually
> prefer Default, not sure why I went with impl. If I can come up with a more
> meaningful name though I'll update it.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/GoogleCloudPlatform/gcloud-intellij/pull/764/files/c86ad100276ba0902fe3d9107b38fcf4e3f5cac3#r69145079>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AHf5He4OERO0qQUQJStF3TXTLs-OrGlZks5qQ9XogaJpZM4JAceh>
> .
>